### ISSUE TITLE
sizeof is used on a pointer when it should refer to its data

### ISSUE THREAD

**Issue by maxcai314** [isotes/tun-open#1]
In tun_open.c, the buffer length is incorrectly generated as the size of the pointer TunOpenName* tunName. Instead, it should reflect the proper size of the buffer.

```c
if (tunName) {
        socklen_t optlen = sizeof(tunName);
        CHK(err, getsockopt(fd, SYSPROTO_CONTROL, UTUN_OPT_IFNAME, tunName, &optlen));
}
```

To fix this, the optlen should be set to `sizeof(tunName->name)`.

**Comment by maxcai314** [comment]
This issue is fixed by pull request #2

**Comment by isotes** [comment]
Thanks for catching this and fixing it!
